lib/services/database/post_upvote_service.dart

```diff
@@ -0,0 +1,131 @@
+import "package:cloud_firestore/cloud_firestore.dart";
+import "package:hooks_riverpod/hooks_riverpod.dart";
+import "package:proxima/models/database/post/post_data.dart";
+import "package:proxima/models/database/post/post_firestore.dart";
+import "package:proxima/models/database/post/post_id_firestore.dart";
+import "package:proxima/models/database/upvote_state.dart";
+import "package:proxima/models/database/user/user_id_firestore.dart";
+import "package:proxima/services/database/firestore_service.dart";
+
+/// This repository service is responsible for handling the upvotes of posts
+class PostUpvoteRepositoryService {
+  /// The name of the subcollection that contains the list of users who upvoted
+  static const upvotersSubCollectionName = "upvoters";
```

Comment:
Is there a reason behind the choice of making two separate subcollections `upvoters` and `downvoters` instead of doing only one called `voters` and have a boolean flag field in it to know if it was an upvote or a downvote ? I feel like the late would be faster (only one request) and probably would facilitate the implementation.

Reply:
My original idea was that I guessed it would simplify my implementation, and that it might make some operations faster because the list we would access would be smaller. After some more reasoning, I think you are right and this is fixed in 003cd5dd0a8097512a10ae604565a947d5c00440.